mlx_graphs/data/collate.py

```diff
@@ -0,0 +1,51 @@
+from typing import List
+
+import mlx.core as mx
+from mlx_graphs.data.data import GraphData
+
+
+def collate(graph_list: List[GraphData]) -> dict:
+    """
+    Collate function to perform the unification of graphs in a batch.
+    By default, it concatenates all default graph attributes in dim 0
+    apart from `edge_index` which is concatenated along dim 1 and
+    nodes indices are incremented
+
+    Args:
+        graph_list (List[GraphData]): the list of GraphData objects to collate
+
+    Returns:
+        dict: dict containing all the attributes of the unified and disconnected big graph as
+        well as the slices corresponding to the portion of node for each graph from the provided
+        list of graphs.
+    """
+    batch_attr_dict = {}
+
+    for attr in graph_list[0].to_dict():
```

Comment:
This is very much an edge case, but what if some graphs in the list have additional attributes? Should they be concatenated in some way? Ignored? Should we throw an error?

Reply:
Ahha that's exactly what I planed to work on today. I think we should either ignore other attributes, or raise an error. 
Just ignoring might by tricky for the user if he's looking for those values in the final batch, and raising might be a bit too strong, but I don't see why some GrahData would have additional attributes.

Reply:
Yeah, I agree that's something that shouldn't happen, at least for homogeneous graphs - it could if they're heterogeneous though, but we can leave that for later!

I think in this case we should raise an error as having extra attributes is something we don't expect, or we can raise a warning saying that extra attributes will be ignored, wdyt?

Reply:
Okay let's go for the raising option!

Reply:
The only thing I'd be slightly worried about is the overhead we'll incur in checking all the graphs in the batch - but it should be a fairly cheap operation.